lib/consensus/isaac.go

```diff
@@ -22,24 +22,25 @@ type ISAAC struct {
 	RunningRounds         map[ /* Round.Hash() */ string]*RunningRound
 	latestConfirmedBlock  block.Block
 	LatestRound           round.Round
-	connectionManager     *network.ConnectionManager
+	connectionManager     network.ConnectionManager
+	proposerSelector      ProposerSelector
```

Comment:
Finally, I added ProposerSelector in ISAAC.
I tried to remove this interface, but it is really complicated because it is related with ISAAC but it should use connectionManager because of validator list.
This is the best way to  that I found.
This interface will be useful when we experiment for the better selector. And we could have modes like sequantial or VRF as ontology.

If some one has any good idea, then please modify it.